docs/rule/role-has-required-attributes.md

```diff
@@ -0,0 +1,27 @@
+# role-has-required-attributes
+
+Elements with ARIA roles must also include all required attributes for that role. This ensures that a given element possesses the necessary states and properties to behave consistently with user expectations for other elements with the same ARIA role.
+
+This rule enforces that elements with an ARIA role also declare all required ARIA attributes for that role.
+
+## Examples
+
+This rule **forbids** the following:
+
+```hbs
+<div role="option" />
+<CustomComponent role="checkbox" aria-required="true" />
+{{some-component role="heading" aria-boguslevel="2"}}
+```
+
+This rule **allows** the following:
+
+```hbs
+<div role="option" aria-selected="false" />
+<CustomComponent role="checkbox" aria-required="true" aria-checked="false" />
+{{some-component role="heading" aria-boguslevel="2" aria-level="2"}}
```

Comment:
```suggestion
{{some-component role="heading" aria-level="2"}}
```

While it is _technically_ allowed by definition (and handled by the `no-invalid-aria-attributes` rule), can we clean up this allowed example to remove the `aria-boguslevel` attribute?

Reply:
Thanks for updating this @judithhinlung! I also dig the updated rule name. 🙌🏽